sources/packages/web/src/components/common/StudentApplication.vue

```diff
@@ -247,12 +248,41 @@ export default {
         event.changed?.component.key === OFFERINGS_DROPDOWN_KEY &&
         +event.changed.value > 0
       ) {
-        await formioComponentLoader.loadSelectedOfferingDate(
+        await formioComponentLoader.loadSelectedOfferingDetails(
           form,
           +event.changed.value,
-          SELECTED_OFFERING_DATE_KEY,
+          {
+            offeringEndDateFieldId: SELECTED_OFFERING_END_DATE_KEY,
+            offeringStartDateFieldId: SELECTED_OFFERING_START_DATE_KEY,
+          },
         );
       }
+      // If the user after selecting a study period finds that
+      // they need to check my study period not listed, then
+      // the details of previously selected
+      // study period must be cleared.
+      if (
```

Comment:
Is there a reason to not handle this logic at the form.io definition directly?
Is it to try to keep the "offering details" for load/reset altogether?

Reply:
Is it to try to keep the "offering details" for load/reset altogether? - Yes the effort is trying to load and reset offering details. 

Side effect when this is not done -  When a offering is selected and then  my program/offering not listed is checked, this variable remains in form submission data with previously selected offering dates, which can potentially cause issues.